### PR TITLE
User may connect a slack team for reward purposes

### PR DESCRIPTION
**Pull request by zspencer**
Resolves https://github.com/wecohere/nourish.party/issues/95

### CODE REVIEW

File: app/views/devise/registrations/edit.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%- content_for :main_class, '--small-centered-content' %>
+<%- content_for :main_class %>
```
Comment (zspencer) [inline app/views/devise/registrations/edit.html.erb]: This page needs to be wider, as it's getting lots more stuff on it than it used to have.

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 OmniAuth.config.full_host = ENV["APP_ROOT_URL"]
 Rails.application.config.middleware.use OmniAuth::Builder do
   provider :stripe_connect, ENV["STRIPE_OAUTH_CLIENT_ID"], ENV["STRIPE_SECRET_KEY"],
-           callback_path: "/auth/stripe_connect/callback"
+           callback_path: "auth/stripe_connect/callback"
+  provider :slack, ENV["SLACK_OAUTH_CLIENT_ID"], ENV["SLACK_OAUTH_CLIENT_SECRET"],
+           scope: "admin,chat:write:bot,team:read,users:read,identify"
```
Comment (zspencer) [inline config/initializers/omniauth.rb]: We'll want to scope this down to what we need when we figure out what we actually need.

File: app/views/devise/registrations/edit.html.erb (hunk 1 above)
Comment (bhaibel) [reply app/views/devise/registrations/edit.html.erb]: If you delete the content-for it should default to the wide layout.

File: app/controllers/slack_teams_controller.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    end
+    redirect_to edit_registered_user_registration_url
+  end
+
+  private def find_or_initialize_slack_team
+    slack_teams.find_or_initialize_by(slack_id: team_data["id"])
+  end
+
+  private def team_params
+    { name: team_data["name"], domain: team_data["domain"],
```
Comment (bhaibel) [inline app/controllers/slack_teams_controller.rb]: Nice job forcing the omniauth data into a Railsy mold!

**Pull request merged**